[DOC] Add documentation for prometheus metrics compatibility (#3320)
grafana/tempo: docs/sources/tempo/metrics-generator/_index.md

To learn more about this processor, read the [documentation]({{< relref "./span_metrics" >}}).

## Local blocks

> The local blocks processor is used to store spans for a set period of time and
> allow for more complex APIs to perform calculations on the data.  This must be
> enabled for certain metrics APIs to function.

Reviewer: ```suggestion
The local blocks processor stores spans for a set period of time and
enables more complex APIs to perform calculations on the data. The processor must be
enabled for certain metrics APIs to function.
```
